pkg/collector/collector.go

```diff
@@ -137,3 +145,51 @@ func (r *ChaosCollector) updateOrCreateEvent(req ctrl.Request, kind string, stat
 
 	return nil
 }
+
+func (r *ChaosCollector) archiveExperiment(req ctrl.Request, obj v1alpha1.InnerObject) error {
+	archive := &core.ArchiveExperiment{
+		ArchiveExperimentMeta: core.ArchiveExperimentMeta{
+			Namespace: req.Namespace,
+			Name:      req.Name,
+			Kind:      obj.GetObjectKind().GroupVersionKind().Kind,
+		},
+	}
+
+	switch chaos := obj.(type) {
+	case *v1alpha1.PodChaos:
+		archive.Action = string(chaos.Spec.Action)
+	case *v1alpha1.NetworkChaos:
+		archive.Action = string(chaos.Spec.Action)
+	case *v1alpha1.IoChaos:
+		archive.Action = string(chaos.Spec.Action)
+	case *v1alpha1.TimeChaos, *v1alpha1.KernelChaos, *v1alpha1.StressChaos:
+		archive.Action = ""
```

Comment:
Maybe we should add some default actions to `TimeChaos`, `KernelChaos` and `StressChaos`

Reply:
@yeya24 How about adding a function `GetAction` into `InnerObject` interface?

Reply:
We can merge this PR first. I will add a common interface today and update this code.